core/src/test/java/io/debezium/connector/cassandra/TestUtils.java

```diff
@@ -89,19 +99,28 @@ public static List<String> getTables(CqlSession session) {
     }
 
     public static void truncateTestKeyspaceTableData() {
+        truncateTestKeyspaceTableData(TEST_KEYSPACE_NAME);
+    }
+
+    public static void truncateTestKeyspaceTableData(String keyspace) {
         try (CqlSession session = CqlSession.builder().build()) {
-            for (String table : getTables(session)) {
-                session.execute(SimpleStatement.newInstance(String.format("TRUNCATE %s.%s", TEST_KEYSPACE_NAME, table)));
+            for (String table : getTables(keyspace, session)) {
+                session.execute(SimpleStatement.newInstance(String.format("TRUNCATE %s.%s", keyspace, table)));
             }
         }
     }
 
-    public static void deleteTestKeyspaceTables() {
+    public static void deleteTestKeyspaceTables() throws Exception {
+        deleteTestKeyspaceTables(TEST_KEYSPACE_NAME);
+    }
+
+    public static void deleteTestKeyspaceTables(String keyspaceName) throws Exception {
         try (CqlSession session = CqlSession.builder().build()) {
-            for (String table : getTables(session)) {
-                session.execute(SimpleStatement.newInstance(String.format("DROP TABLE IF EXISTS %s.%s", TEST_KEYSPACE_NAME, table)));
+            for (String table : getTables(keyspaceName, session)) {
+                session.execute(SimpleStatement.newInstance(String.format("DROP TABLE IF EXISTS %s.%s", keyspaceName, table)));
             }
         }
+        Thread.sleep(5000);
```

Comment:
Why is the sleep needed here? Could it be replaced by a condition to check?

Reply:
it is there for schema listener in debezium to have a chance to react on the schema changes in Cassandra. 5s on keyspace creation seems to be like a safe value. I do not think waiting by a "condition" is appropriate as I do not know what that condition would actually wait for in this context.

Reply:
ok, I understand what's missing/is difference from what I expected. In case of RDBMS we have `DatabaseHistoryMetrics` that is set of JMX metrics allowing user to monitor schema changes. Maybe Cassandra should provide a similar instrument. In that case it would be also pretty easy to check the metrics and wait for change to be reacted on.

Reply:
probably, we might do this on top of the current pr after it is merged, I do not think the change you are suggesting is related to this PR as such.

Reply:
Aggred, there is follow-up Jira for that - https://issues.redhat.com/browse/DBZ-5523